src/main/java/nwoolcan/controller/brewery/BreweryControllerImpl.java

```diff
@@ -20,8 +37,71 @@ public BreweryControllerImpl(final Brewery brewery) {
         this.warehouseController = new WarehouseControllerImpl(brewery.getWarehouse());
     }
 
+    @Override
+    public Collection<MasterBatchViewModel> getBatches(final QueryBatch queryBatch) {
+        return brewery.getBatches(queryBatch)
+                      .stream()
+                      .map(MasterBatchViewModel::new)
+                      .collect(Collectors.toList());
+    }
+
+    @Override
+    public Result<Empty> stockBatch(final int batchId, final int beerArticleId, final Date expirationDate) {
+        final Result<Batch> batchResult = getBatchById(batchId);
+        final Result<BeerArticle> beerArticleResult = getBeerArticleById(beerArticleId);
+        return Result.ofEmpty()
+                     .flatMap(() -> batchResult)
+                     .flatMap(() -> beerArticleResult)
+                     .flatMap(() -> brewery.stockBatch(batchResult.getValue(),
+                                                       beerArticleResult.getValue(),
+                                                       expirationDate));
+    }
+
+    @Override
+    public Result<Empty> stockBatch(final int batchId, final int beerArticleId) {
+        final Result<Batch> batchResult = getBatchById(batchId);
```

Comment:
```suggestion
        return getBatchById(batchId).flatMap(batchRes ->
                                                getBeerArticleById(beerArticleId)
                                                    .map(beerArticleRes -> Pair.of(batchRes, beerArticleRes)))
                                    .flatMap(pair -> brewery.stockBatch(pair.getLeft(), pair.getRight()));
```

Reply:
Something like that?

Reply:
I already thought at that solution buy, IMHO, it is more clear in my way. I just find it clearer and smoother. I can agree it may is better to call the first function to start, but still it looks more uniform checking all the independent result one by one.